Lib/test/test_selectors.py

```diff
@@ -399,17 +399,18 @@ def handler(*args):
 
         orig_alrm_handler = signal.signal(signal.SIGALRM, handler)
         self.addCleanup(signal.signal, signal.SIGALRM, orig_alrm_handler)
-        self.addCleanup(signal.alarm, 0)
 
         signal.alarm(1)
-
-        s.register(rd, selectors.EVENT_READ)
-        t = time()
-        # select() is interrupted by a signal which raises an exception
-        with self.assertRaises(InterruptSelect):
-            s.select(30)
-        # select() was interrupted before the timeout of 30 seconds
-        self.assertLess(time() - t, 5.0)
+        try:
+            s.register(rd, selectors.EVENT_READ)
```

Comment:
What if the process is switched before this line, paused on more than 1 sec, and a SIGALRM signal is raised before registering a selector?

Reply:
I moved the signal.alarm(1) into the try/except.

"

"What if  (...) a SIGALRM signal is raised before registering a selector?" : the test would fail, no? In my PR, I only care of making sure that a scheduled alarm is always cancelled if something goes wrong.